cnn.py

```diff
@@ -1,198 +1,255 @@
 import tensorflow as tf
 import numpy as np
 import TRAINING_VARIABLES
+from data import DataSet
+
+
+'''
+MODULE FUNCTIONS
+'''
+
+
+def weight_variable(shape, name):
+    initial = tf.truncated_normal(shape, stddev=0.1)
+    return tf.Variable(initial, name=name)
+
+
+def bias_variable(shape, name):
+    initial = tf.constant(0.1, shape=shape)
+    return tf.Variable(initial, name=name)
+
+
+def conv2d(x, W, filter_type):
+    return tf.nn.conv2d(x, W, strides=[1, 1, 1, 1], padding=filter_type)
+
+
+def max_pool_2x2(x):
+    return tf.nn.max_pool(x, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
+
+
+def get_conv_layer(input_variables, kernels_in, kernels_out, layer_number, filter_x, filter_y, filter_type, name):
+    weights_shape = [filter_y, filter_x, kernels_in, kernels_out]
+    weights_name = name + "w_conv_" + layer_number
+    weights = weight_variable(weights_shape, weights_name)
+
+    bias_shape = [kernels_out]
+    bias_name = name + 'b_conv_' + layer_number
+    bias = bias_variable(bias_shape, bias_name)
+
+    features = conv2d(input_variables, weights, filter_type) + bias
+    return tf.nn.relu(features)
+
+
+def get_nn_layer(input_variables, connections_in, connections_out, layer_number, name):
+    weights_shape = [connections_in, connections_out]
+    weights_name = name + 'w_fc_' + str(layer_number + 1)
+    weights = weight_variable(weights_shape, weights_name)
+
+    bias_shape = [connections_out]
+    bias_name = name + 'b_fc_' + str(layer_number + 1)
+    bias = bias_variable(bias_shape, bias_name)
+
+    features = tf.matmul(input_variables, weights) + bias
+    return tf.nn.relu(features)
+
+
+'''
+CONFIGURATION VARIABLES
+'''
+
 
 V = TRAINING_VARIABLES.VARS()
 
 
-class Convolutional_Neural_Network(object):
+'''
+CNN CLASS
+'''
+
+
+class ConvolutionalNeuralNetwork(object):
     def __init__(self):
```

Comment:
A lot of the variables could (and should) be passed to the net as it is created rather than hard-loaded from V. This can be done in a later branch, but just for the record.